### ISSUE TITLE
update_attributes does not support passing in ActionController::Parameters

### ISSUE THREAD

**Issue by jmcnevin** [rails/activeresource#245]
In attempting to upgrade a project to rails 5, I noticed that in those places where I'm using `update_attributes`, I receive an exception when passing in the params, if I don't first call `to_h`.

```
expected an attributes Hash, got <ActionController::Parameters {...} permitted: true>
```

**Comment by arich** [comment]
Hello @rafaelfranca! We're still experiencing this issue. We've had to patch our `base_resource.rb` with the following code:
```
  def load(attributes, remove_root = false, persisted = false)
    if !attributes.is_a?(Hash) && attributes.respond_to?(:to_h)
      attributes = attributes.to_h
    end
    super(attributes, remove_root, persisted)
  end
```

The issue with the code added [here](https://github.com/rails/activeresource/commit/3b74650ffa319a4ab1e4739c342720a8eeaad16f#diff-36351a602aa8104f544d1fe1f9fa34d2R1388), is that an instance of `ActiveController::Parameters` responds to `to_h`, but does not respond to `to_hash`. 

```
irb(main):001:0> params_instance = ActionController::Parameters.new
=> <ActionController::Parameters {} permitted: false>
irb(main):002:0> params_instance.to_hash
=> {}
irb(main):003:0> params_instance.to_h
=> {}
irb(main):004:0> params_instance.respond_to?(:to_h)
=> true
irb(main):005:0> params_instance.respond_to?(:to_hash)
=> false
```

Can this be reopened?

**Comment by arich** [comment]
Nevermind, it seems this was fixed in Rails 5.0.4. Thanks!
